woocommerce/woocommerce-ios
# Add-Ons: Create view to render order add-ons

closes #3893

# Why

This PR makes sure the user can navigate from the order item cell to a screen where the product add-ons are rendered in a very simple list.

Note: This is my first PR using `SwiftUI` and I'm trying to put what I consider "best" practices(from my limited experience) here: paNNhX-cY-p2.

# How

- Updated `OrderDetailsDataSource` to provide the necessary callback when the "View add-ons" button is pressed with the correct add-on attributes.

- Created a `WooStyleModifiers` file to migrate functions from `UILabel+Helpers` and `UIButton+Helpers` as they are needed in `SwiftUI` views.

- Created `OrderAddOnListI1View` which is a `SwiftUI` view that renders one or more `OrderAddOnI1View` views based on it's view model content.

- Created `OrderAddOnListI1ViewModel` which is in charge of receiving order item attributes and converting them to view models that can be consumed by the `OrderAddOnListI1View`. Here the workaround magic happens. 

- Created `OrderAddOnI1ViewModel` which represents a single add-on, to be consumed by `OrderAddOnListI1View`.

- Created `OrderAddOnsListViewController` as a hosting controller of `OrderAddOnListI1View`

- Updated `OrderDetailsViewController` to launch the newly created `OrderAddOnsListViewController`

# Screenshots

Light | Dark | Big Font
--- | --- | ---
<img width="396" alt="regular" src="https://user-images.githubusercontent.com/562080/114953647-696eab00-9e1e-11eb-9dcd-3f8a0cb5d69a.png"> | <img width="406" alt="dark" src="https://user-images.githubusercontent.com/562080/114953644-68d61480-9e1e-11eb-84ee-5e01111ad7f7.png"> | <img width="411" alt="big font" src="https://user-images.githubusercontent.com/562080/114953642-67a4e780-9e1e-11eb-89d7-0b7c355b962a.png">


# Testing Steps

**Prerequisites:** Have your site with the add-ons plugin installed and configured.

- Make an order with an item with add-ons
- Launch the app and navigate to that order
- Tap on the "View Add-Ons" button.
- See that a new screen is presented with the add-on's information.

Update release notes:
